eng/pipelines/templates/stages/archetype-python-release.yml

```diff
@@ -103,28 +116,15 @@ stages:
                 runOnce:
                   deploy:
                     steps:
-                      - checkout: self
-
-                      - download: current
-                        artifact: ${{parameters.ArtifactName}}
```

Comment:
Do we not still need this download step?

Reply:
Negative! It's taken care of by the `templateContext` where we set up the input artifact 👍 Line 99 -> 109